### ISSUE TITLE
[21.0.0] userCanceled isn't true

### ISSUE THREAD

**Issue by Bardiamist** [onfido/onfido-ios-sdk#224]
### What was the expected behaviour?

`userCanceled` should be true

### What happened instead?

`userCanceled` isn't true

### Version info:

- Cocoapods (if applicable): 1.10.1
- Onfido SDK (Debug/Release): 21.0.0
- iOS: 14.4
- Xcode: Version 12.4 (12D4e)
- Device/Simulator: Simulator
- Device/Simulator language: English

### Integration configuration:

```
      ONFlow *onFlow = [[ONFlow alloc] initWithFlowConfiguration:config];
      [onFlow withResponseHandler:^(ONFlowResponse *response) {
        if (response.userCanceled) {
          NSLog(@"That never be logged");
        }
      }];
```

I can provide full code if need

### Steps to reproduce:

Present onfidoController
Close onfidoController

### Additional info:

No problems with v`20.1.0`.

**Comment by Bardiamist** [comment]
Not fixed in 21.1.0

**Comment by anuragajwani** [comment]
@Bardiamist sorry for the late reply.

On version 21.0.0 we changed `userCanceled` property from `Bool` to `CancellationReason` enum.

There are two reasons why userCanceled can be called at the moment:
1. flow exit
2. denied consent (when consent screen configured, shown and user does not consent)

To adapt to the new breaking change please change to code to:

```
if (flowResponse.userCanceled == CancellationReasonUserExit || flowResponse.userCanceled == CancellationReasonDeniedConsent) {
    NSLog(@"This should be logged");
}
```

This was the main reason for the breaking change. We didn't appropriately update the Objective-C code though. 

Sorry for the inconvenience caused. We will update the documentation.

Anurag

**Comment by anuragajwani** [comment]
I just noticed that the value would always be set regardless whether or not the user cancelled. I have raised the bug.

**Comment by anuragajwani** [comment]
@Bardiamist we have issued a fix and will be released soon

**Comment by anuragajwani** [comment]
@Bardiamist the fix has been released in [21.2.0](https://github.com/onfido/onfido-ios-sdk/blob/master/CHANGELOG.md#2120---2021-03-29). We went with the least breaking change approach so you can keep your code:
```
if (response.userCanceled) {
        // Flow cancelled by the user
        // You can check why the user canceled using response.userCanceled.reason
} else ...
```

`userCanceled` will now be set to `NULL` when not exited by user and when set you can find the reason why by querying the reason property within the user canceled object:
```
response.userCanceled.reason
```

Please let me know if your issue is fixed.

**Comment by Bardiamist** [comment]
Looks good
